core/sched.c

```diff
@@ -181,6 +190,12 @@ void sched_switch(uint16_t other_prio)
 
 NORETURN void sched_task_exit(void)
 {
+    /* Prevent optimization of these symbols */
+    max_threads = max_threads;
```

Comment:
This will cause warnings from various static analyzers. Does adding `__attribute__((used))` help?